python/lsst/cp/pipe/cpSkyTask.py

```diff
@@ -0,0 +1,417 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import numpy as np
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+
+from lsst.pipe.drivers.background import (FocalPlaneBackground, MaskObjectsTask, SkyMeasurementTask,
+                                          FocalPlaneBackgroundConfig)
+from lsst.daf.base import PropertyList
+from ._lookupStaticCalibration import lookupStaticCalibration
+from .cpCombine import CalibCombineTask
+
+__all__ = ['CpSkyImageTask', 'CpSkyImageConfig',
+           'CpSkyScaleMeasureTask', 'CpSkyScaleMeasureConfig',
+           'CpSkySubtractBackgroundTask', 'CpSkySubtractBackgroundConfig',
+           'CpSkyCombineTask', 'CpSkyCombineConfig']
+
+
+class CpSkyImageConnections(pipeBase.PipelineTaskConnections,
+                            dimensions=("instrument", "physical_filter", "exposure", "detector")):
+    inputExp = cT.Input(
+        name="cpSkyIsr",
+        doc="Input pre-processed exposures to combine.",
+        storageClass="Exposure",
+        dimensions=("instrument", "exposure", "detector"),
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        doc="Input camera to use for geometry.",
+        storageClass="Camera",
+        dimensions=("instrument",),
+        lookupFunction=lookupStaticCalibration,
+        isCalibration=True,
+    )
+
+    maskedExp = cT.Output(
+        name="cpSkyMaskedIsr",
+        doc="Output masked post-ISR exposure.",
+        storageClass="Exposure",
+        dimensions=("instrument", "exposure", "detector"),
+    )
+    maskedBkg = cT.Output(
+        name="cpSkyDetectorBackground",
+        doc="Initial background model from one image.",
+        storageClass="FocalPlaneBackground",
+        dimensions=("instrument", "exposure", "detector"),
+    )
+
+
+class CpSkyImageConfig(pipeBase.PipelineTaskConfig,
+                       pipelineConnections=CpSkyImageConnections):
+    maskTask = pexConfig.ConfigurableField(
+        target=MaskObjectsTask,
+        doc="Object masker to use.",
+    )
+
+    maskThresh = pexConfig.Field(
+        dtype=float,
+        default=3.0,
+        doc="k-sigma threshold for masking pixels.",
+    )
+    maskList = pexConfig.ListField(
+        dtype=str,
+        default=["DETECTED", "BAD", "NO_DATA", "SAT"],
+        doc="Mask planes to reject.",
+    )
+
+    largeScaleBackground = pexConfig.ConfigField(
+        dtype=FocalPlaneBackgroundConfig,
+        doc="Large-scale background configuration",
+    )
+
+    def setDefaults(self):
+        # self.largeScaleBackground.xSize = 256
+        # self.largeScaleBackground.ySize = 256
```

Comment:
Why are these inline comments necessary? Are they hinting at recommended defaults?

Reply:
I've removed the `256` comments, and clarified the `obs_subaru` comment.  As these values are not completely intuitive (very few other places in the code use mm), I wanted to note where they came from.